reactor-core/src/main/java/reactor/core/publisher/FluxOnBackpressureBuffer.java

```diff
@@ -349,7 +364,29 @@ public boolean isEmpty() {
 
 		@Override
 		public void clear() {
-			Operators.onDiscardQueueWithClear(queue, ctx, null);
+			// use guard on the queue instance as the best way to ensure there is no racing on draining
+			// the call to this method must be done only during the ASYNC fusion so all the callers will be waiting
+			// this should not ber performance costly with the assumption the cancel is rare operation
+			if (DISCARD_GUARD.getAndIncrement(this) != 0) {
+				return;
+			}
+
+			int missed = 1;
+
+			for (;;) {
+				Operators.onDiscardQueueWithClear(queue, ctx, null);
+
+				int w = discardGuard;
```

Comment:
nitpick: can you rename the variable to at least stick to the convention that local copies in drain loops are named after the initial of their corresponding field? it. `w` becomes `dg`

Reply:
applies to all occurrences below as well

Reply:
fixed